frontend/test/pytest/test_verification.py

```diff
@@ -206,6 +217,96 @@ def f(x: float):
                 return qml.expval(qml.PauliX(0))
 
 
+class TestObservableValidation:
+    """Tests the general validation of observables (independent of gradient method)"""
+
+    def test_unsupported_observable_raises_error(self):
+        """Test that including an unsupported observable in a measurement raises an
+        error when jitting the circuit"""
+
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qml.qnode(dev)
+        def f():
+            qml.RX(1.23, 0)
+            return qml.expval(qml.RX(1.2, 0))
+
+        with pytest.raises(CompileError, match="RX.*not supported as an observable"):
+            qml.qjit(f)()
+
+    @pytest.mark.parametrize(
```

Comment:
Maybe add a few cases for nested `Prod` and `SProd`?

Reply:
added